shell/edit/secret/index.vue

```diff
@@ -312,22 +393,46 @@ export default {
     <CruResource
       v-else
       :mode="mode"
-      :validation-passed="true"
+      :validation-passed="fvFormIsValid"
       :selected-subtype="value._type"
       :resource="value"
       :errors="errors"
-      :done-route="doneRoute"
       :subtypes="secretSubTypes"
+      :cancel-event="true"
       @finish="saveSecret"
       @select-type="selectType"
       @error="e=>errors = e"
+      @cancel="redirectAfterCancel"
     >
       <NameNsDescription
+        v-if="!isProjectScoped"
         :value="value"
         :mode="mode"
         :namespaced="!isCloud"
         @update:value="$emit('input', $event)"
       />
+      <NameNsDescription
```

Comment:
I think we may want to create a NamePorjectDescription just to make the usage more clear. Up to you though.